components/select/Select.razor.cs

```diff
@@ -490,6 +490,13 @@ internal bool IsGroupingEnabled
 
         #endregion Properties
 
+        private static readonly bool _isItemAndValueSameType;
+
+        static Select()
```

Comment:
Why is this static?

Reply:
Of course this can be static. All instances of `Select<T1, T1>` will share the same `_isItemAndValueSameType`. But instances of `Select<T2, T2>` will use another `_isItemAndValueSameType`. And for generic class static fields, the runtime will distribute memory when first time use the specific generic class .

Reply:
Of course this can be static. All instances of `Select<T1, T1>` will share the same `_isItemAndValueSameType`. But instances of `Select<T2, T2>` will use another `_isItemAndValueSameType`. And for generic class static fields, the runtime will distribute memory when first time use the specific generic class .

Reply:
Now I want to just remove this judgement for `!_isItemAndValueSameType && string.IsNullOrWhiteSpace(ValueName)`. As you said, it is possible to use expression tree to build the delegate. So `null` or `string.Empty` or whitespace should be valid value for `ValueName` .